File: 0712-minimum-ascii-delete-sum-for-two-strings/0712-minimum-ascii-delete-sum-for-two-strings.cpp
class Solution {
public:
    int minimumDeleteSum(string s1, string s2) {
        vector<vector<int>> all(s1.size()+1,vector<int>(s2.size()+1));
        int answer=0;
        for(int i=1; i<=s1.size(); i++){
            for(int j=1; j<=s2.size(); j++){
                if(s1[i-1]==s2[j-1])
                    all[i][j] = all[i-1][j-1]+s1[i-1];
                else
                    all[i][j] = max(all[i-1][j], all[i][j-1]);
            }
        }
        
        for(auto e:s1) answer+=e;
        for(auto e:s2) answer+=e;
        return answer-2*all[s1.size()][s2.size()];
    }
};

File: 0712-minimum-ascii-delete-sum-for-two-strings/0712-minimum-ascii-delete-sum-for-two-strings.py
class Solution:
    def minimumDeleteSum(self, s1: str, s2: str) -> int:
        alls=[]
        sub=[]
        answer=0
        n1 = len(s1)
        n2 = len(s2)
        for i in range(n2+1):
            sub.append(0)
        alls.append(sub)
        for i in range(1,n1+1):
            alls.append([0]);
            for j in range(1,n2+1):
                if s1[i-1]==s2[j-1]:
                    alls[i].append(alls[i-1][j-1]+ord(s1[i-1]))
                else:
                    alls[i].append(max(alls[i-1][j], alls[i][j-1]))
        for i in range(n1):
            answer+=ord(s1[i])
        for i in range(n2):
            answer+=ord(s2[i])
        answer-=2*alls[n1][n2]
        return answer